From ea530483ad42e5b9a89934fbdfd57df81b3a90ab Mon Sep 17 00:00:00 2001
From: John Weber <rjohnweber@gmail.com>
Date: Fri, 15 Mar 2013 08:43:15 -0500
Subject: [PATCH] wandboard: fix sdhc platform data

Fixes sdhc platfrom data structure to solve a problem when loading
the firmware files for the on-board Wifi module on Wandboard-dual.
This problem only occured when udev is tasked with loading firmware
from the filesystem.

Upstream-Status: Pending

Signed-off-by: John Weber <rjohnweber@gmail.com>
---
 arch/arm/mach-mx6/board-wand.c |   11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx6/board-wand.c b/arch/arm/mach-mx6/board-wand.c
index 7f972eb..7be6726 100644
--- a/arch/arm/mach-mx6/board-wand.c
+++ b/arch/arm/mach-mx6/board-wand.c
@@ -184,13 +184,17 @@ static const struct esdhc_platform_data wand_sd_data[3] = {
 		.cd_gpio		= WAND_SD1_CD,
 		.wp_gpio		=-EINVAL,
 		.keep_power_at_suspend	= 1,
	        .support_8bit		= 0,
+	  .delay_line = 0,
 		.platform_pad_change	= plt_sd_pad_change,
+		.cd_type = ESDHC_CD_CONTROLLER,
 	}, {
-		.cd_gpio		=-EINVAL,
-		.wp_gpio		=-EINVAL,
 		.keep_power_at_suspend	= 1,
 		.platform_pad_change	= plt_sd_pad_change,
+		.always_present = 1,
+		.support_8bit = 0,
+		.delay_line = 0,
+		.cd_type = ESDHC_CD_PERMANENT,
 	}, {
 		.cd_gpio		= WAND_SD3_CD,
 		.wp_gpio		= WAND_SD3_WP,
@@ -198,6 +202,7 @@ static const struct esdhc_platform_data wand_sd_data[3] = {
 		.support_8bit		= 0,
 		.delay_line		= 0,
 		.platform_pad_change	= plt_sd_pad_change,
+		.cd_type = ESDHC_CD_CONTROLLER,
 	}
 };
 
-- 
1.7.9.5

